lib/rqrcode/qrcode/qr_code.rb

```diff
@@ -117,7 +129,8 @@ def initialize( string, *args )
 
       @data                 = string
 
-      mode                  = QRAlphanumeric.valid_data?( @data ) ? :mode_alpha_numk : :mode_8bit_byte
+      mode                  = QRMODE_NAME[(options[:mode] || '').to_sym]
```

Comment:
You are quite right. I haven't written much during the last two years so i have missed that. Fix the above or just remove 1.8 from .travis.yml.